### PR TITLE
on observation update, remove problematic 'version' string and don't require .properties

### PR DESCRIPTION
**Pull request by karissa**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -132,11 +132,8 @@ Api.prototype.observationUpdate = function (req, res, m) {
         })
         opts.links = old.id
       }
-      var newObs = Object.assign(old, {
-        properties: obs.properties,
-        lat: obs.lat,
-        lon: obs.lon
-      })
+      var newObs = Object.assign(old, {})
```
Comment (czengg) [inline api.js]: I'm confused. Is old the observation that is in the database? How does it get updated with what gets passed in? Seems like you're assigning newObs to be the old db observation with the only update being the removal of the version? Unless I'm reading this wrong...

Comment (karissa) [reply api.js]: ya you're right I should have had more coffee before doing this

File: api.js [hunk 2]
```diff
@@ -132,11 +132,8 @@ Api.prototype.observationUpdate = function (req, res, m) {
         })
         opts.links = old.id
       }
-      var newObs = Object.assign(old, {
-        properties: obs.properties,
-        lat: obs.lat,
-        lon: obs.lon
-      })
+      var newObs = Object.assign(old, m, {})
+      delete newObs.version
```
Comment (noffle) [inline api.js]: Could you delete `newObs.id` as well?

Comment (noffle) [reply api.js]: Oh, and add a test that sets both & ensures they aren't overriding the true version/id?

File: api.js [hunk 3]
```diff
@@ -132,11 +132,8 @@ Api.prototype.observationUpdate = function (req, res, m) {
         })
         opts.links = old.id
       }
-      var newObs = Object.assign(old, {
-        properties: obs.properties,
```
Comment (gmaclennan) [inline api.js]: `m` is the match of parameters, not the observation. `obs` is the new observation from the client.

Comment (karissa) [reply api.js]: ok fixed

File: api.js (hunk 1 above)
Comment (noffle) [reply api.js]: Woah! Very troubling if the tests were passing!

File: api.js [hunk 4]
```diff
@@ -9,6 +9,8 @@ var xtend = require('xtend')
 
 module.exports = Api
 
+var VALID_PROPS = ['lon', 'lat', 'attachments', 'tags']
```
Comment (noffle) [inline api.js]: `s/VALID_PROPS/VALID_UPDATE_PROPS`

Comment (noffle) [reply api.js]: Just because it's a global & the contextual clue could be helpful.

File: api.js [hunk 5]
```diff
@@ -132,11 +134,9 @@ Api.prototype.observationUpdate = function (req, res, m) {
...
         opts.links = old.id
       }
-      var newObs = Object.assign(old, {
-        properties: obs.properties,
-        lat: obs.lat,
-        lon: obs.lon
-      })
+      var newObs = Object.assign(old, {})
+      VALID_PROPS.forEach(function (prop) { newObs[prop] = obs[prop] })
+      delete newObs.version
```
Comment (noffle) [inline api.js]: If we're using a whitelist, this line can be dropped.

**Comment by czengg** [comment]
@karissa do you have an eta for this PR?

**Comment by karissa** [comment]
if anyone has time to pick this up and write some tests, that would be helpful. I'm traveling and trying to get mapeo desktop into a stable state with the new mapfilter changes!

**Comment by noffle** [comment]
Possibly superceded by https://github.com/digidem/mapeo-server/pull/22

**Pull request closed without merging**